### PR TITLE
Issue #384 cisco_asa_show_route - Fix uptime issue

### LINKED ISSUE #384: Cisco ASA show route template error
##### ISSUE TYPE

 - Bug Report

##### TEMPLATE USING

```
Value Filldown PROTOCOL (C|S|R|B|D|O|i|L)
Value Filldown TYPE (\w{0,2})
Value Required,Filldown NETWORK (\d+\.\d+\.\d+\.\d+|\S+)
Value Filldown MASK (\d+\.\d+\.\d+\.\d+)
Value DISTANCE (\d+)
Value METRIC (\d+)
Value NEXTHOPIP (\d+\.\d+\.\d+\.\d+)
Value NEXTHOPIF (\S+)
Value UPTIME (\d+:\d+:\d+)

Start
  ^Gateway -> ROUTES

ROUTES
  # Match regular routes with all data in same line
  ^${PROTOCOL}(\s|\*)${TYPE}\s+${NETWORK}\s+${MASK}\s\[${DISTANCE}\/${METRIC}\]\svia\s${NEXTHOPIP}(,\s${UPTIME}){0,1}(,\s${NEXTHOPIF}){0,1} -> Record
  #
  # Clear all non Filldown variables when line started with network that is variably subnetted
  ^\s+[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}${MASK} -> Clear
  ^${PROTOCOL}(\s|\*)${TYPE}\s+${NETWORK}\s\[${DISTANCE}\/${METRIC}\]\svia\s${NEXTHOPIP}(,\s${UPTIME}){0,1}(,\s${NEXTHOPIF}){0,1} -> Record
  #
  # Match load-balanced routes
  ^\s+\[${DISTANCE}\/${METRIC}\]\s+via\s+${NEXTHOPIP},\s+${NEXTHOPIF} -> Record
  #
  # Match directly connected routes
  ^${PROTOCOL}\s${TYPE}\s+${NETWORK}\sis\sdirectly\sconnected,\s${NEXTHOPIF} -> Record
  ^${PROTOCOL}(\*){0,1}\s${TYPE}\s+${NETWORK}\s+${MASK}\sis\sdirectly\sconnected,\s${NEXTHOPIF} -> Record
  #
  # Clear all variables on empty lines
  ^\s* -> Clearall
  #^${TYPE} -> Continue.Record
  #^${TYPE}\s+${NETWORK}\s+${MASK}\s+\[\d+\/\d+\]\s+via\s+${GATEWAY}\,\s+${INTFC}\s* -> Record
  #^\s+\[\d+\/\d+\]\s+via\s+${GATEWAY}\,\s+${INTFC} -> Record

EOF
```

##### SAMPLE COMMAND OUTPUT

```
Routing Table: BUSINESS
Codes: L - local, C - connected, S - static, R - RIP, M - mobile, B - BGP
       D - EIGRP, EX - EIGRP external, O - OSPF, IA - OSPF inter area
       N1 - OSPF NSSA external type 1, N2 - OSPF NSSA external type 2
       E1 - OSPF external type 1, E2 - OSPF external type 2, V - VPN
       i - IS-IS, su - IS-IS summary, L1 - IS-IS level-1, L2 - IS-IS level-2
       ia - IS-IS inter area, * - candidate default, U - per-user static route
       o - ODR, P - periodic downloaded static route, + - replicated route
Gateway of last resort is 10.200.0.2 to network 0.0.0.0

D EX     10.118.64.0 255.255.255.0 [170/3840] via 10.200.0.2, 2w2d, TRUST
                                 [170/3840] via 10.200.0.1, 2w2d, TRUST
```

##### SUMMARY

The UPTIME variable is not correctly placed in the template, and is causing uptime to populate into nexthopif, and uptime to be blank.

##### STEPS TO REPRODUCE

```
from netmiko import Netmiko

params = {
    'ip': 'testdevice',
    'device_type': 'cisco_asa',
    'username': 'testuser',
    'password': 'testpass',
    'secret': 'testpass'
}

device = Netmiko(**params)

routes = device.send_command('show route', use_textfsm=True)

print(routes)

```

##### EXPECTED RESULTS

```
{
            "protocol": "D",
            "type": "EX",
            "network": "10.118.64.0",
            "mask": "255.255.255.0",
            "distance": "170",
            "metric": "3840",
            "nexthopip": "10.200.0.1",
            "nexthopif": "TRUST",
            "uptime": "3w1d",
        }
```
##### ACTUAL RESULTS

```
{
            "protocol": "D",
            "type": "EX",
            "network": "10.118.64.0",
            "mask": "255.255.255.0",
            "distance": "170",
            "metric": "3840",
            "nexthopip": "10.200.0.1",
            "nexthopif": "3w1d",
            "uptime": "",
        }
```

### PR DESCRIPTION
**Pull request by brandomando**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT

cisco_asa_show_route
##### SUMMARY

Uptime populating into nexthopif value because the uptime value was not placed into the template correctly. 

Fixes #384 

Before:
```
[{'protocol': 'S',
  'type': '',
  'network': 'Server1',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.64.65.82',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.100.67.246',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': 'Sever-2',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '192.168.14.95',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': 'Server_3',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.54.6.0',
  'mask': '255.255.255.0',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': 'SERVER',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '192.168.10.0',
  'mask': '255.255.255.64',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': 'server',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.14.128.0',
  'mask': '255.255.255.0',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.13.0.0',
  'mask': '255.255.0.0',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.3.18.16',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'C',
  'type': '',
  'network': '192.168.1.0',
  'mask': '255.255.255.0',
  'distance': '',
  'metric': '',
  'nexthopip': '',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'L',
  'type': '',
  'network': '192.168.1.108',
  'mask': '255.255.255.255',
  'distance': '',
  'metric': '',
  'nexthopip': '',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'O',
  'type': 'IA',
  'network': '172.16.2.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.2.2',
  'nexthopif': 'inside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '172.16.3.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.2.2',
  'nexthopif': 'inside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '172.16.1.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.2.2',
  'nexthopif': 'inside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '172.16.4.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.2.2',
  'nexthopif': 'inside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '10.0.4.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '10.0.1.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '10.0.2.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '10.0.3.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'E2',
  'network': '10.1.1.0',
  'mask': '255.255.255.0',
  'distance': '170',
  'metric': '3840',
  'nexthopip': '192.168.1.2',
  'nexthopif': '3:21:21',
  'uptime': ''},
 {'protocol': 'O',
  'type': 'E2',
  'network': '10.1.2.0',
  'mask': '255.255.255.0',
  'distance': '170',
  'metric': '3840',
  'nexthopip': '192.168.1.2',
  'nexthopif': '3:21:21',
  'uptime': ''},
 {'protocol': 'O',
  'type': 'E2',
  'network': '10.1.3.0',
  'mask': '255.255.255.0',
  'distance': '170',
  'metric': '3840',
  'nexthopip': '192.168.1.2',
  'nexthopif': '3:21:21',
  'uptime': ''},
 {'protocol': 'C',
  'type': '',
  'network': '192.168.1.0',
  'mask': '255.255.255.0',
  'distance': '',
  'metric': '',
  'nexthopip': '',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'C',
  'type': '',
  'network': '192.168.2.0',
  'mask': '255.255.255.0',
  'distance': '',
  'metric': '',
  'nexthopip': '',
  'nexthopif': 'inside',
  'uptime': ''}]
```
After:
```
[{'protocol': 'S',
  'type': '',
  'network': 'Server1',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.64.65.82',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.100.67.246',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': 'Sever-2',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '192.168.14.95',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': 'Server_3',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.54.6.0',
  'mask': '255.255.255.0',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': 'SERVER',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '10.0.5.12',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '192.168.10.0',
  'mask': '255.255.255.64',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': 'server',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.14.128.0',
  'mask': '255.255.255.0',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.13.0.0',
  'mask': '255.255.0.0',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'S',
  'type': '',
  'network': '10.3.18.16',
  'mask': '255.255.255.255',
  'distance': '1',
  'metric': '0',
  'nexthopip': '172.52.64.1',
  'nexthopif': 'inside',
  'uptime': ''},
 {'protocol': 'C',
  'type': '',
  'network': '192.168.1.0',
  'mask': '255.255.255.0',
  'distance': '',
  'metric': '',
  'nexthopip': '',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'L',
  'type': '',
  'network': '192.168.1.108',
  'mask': '255.255.255.255',
  'distance': '',
  'metric': '',
  'nexthopip': '',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'O',
  'type': 'IA',
  'network': '172.16.2.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.2.2',
  'nexthopif': 'inside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '172.16.3.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.2.2',
  'nexthopif': 'inside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '172.16.1.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.2.2',
  'nexthopif': 'inside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '172.16.4.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.2.2',
  'nexthopif': 'inside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '10.0.4.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '10.0.1.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '10.0.2.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'IA',
  'network': '10.0.3.254',
  'mask': '255.255.255.255',
  'distance': '110',
  'metric': '11',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'E2',
  'network': '10.1.1.0',
  'mask': '255.255.255.0',
  'distance': '170',
  'metric': '3840',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'E2',
  'network': '10.1.2.0',
  'mask': '255.255.255.0',
  'distance': '170',
  'metric': '3840',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'O',
  'type': 'E2',
  'network': '10.1.3.0',
  'mask': '255.255.255.0',
  'distance': '170',
  'metric': '3840',
  'nexthopip': '192.168.1.2',
  'nexthopif': 'outside',
  'uptime': '3:21:21'},
 {'protocol': 'C',
  'type': '',
  'network': '192.168.1.0',
  'mask': '255.255.255.0',
  'distance': '',
  'metric': '',
  'nexthopip': '',
  'nexthopif': 'outside',
  'uptime': ''},
 {'protocol': 'C',
  'type': '',
  'network': '192.168.2.0',
  'mask': '255.255.255.0',
  'distance': '',
  'metric': '',
  'nexthopip': '',
  'nexthopif': 'inside',
  'uptime': ''}]
```

### CODE REVIEW

**Comment by brandomando** [comment]
Sorry for the sloppy commits. Please feel free to squash merge if you choose to merge. Let me know if there are any comments/fixes that need to be looked at with this, but the issue was fairly straightforward. The value `UPTIME` was missing in several places, and there was a space prior to `\s+` in the load-balanced template.

Thanks,

Brandon

**Comment by brandomando** [comment]
Also fixed, the uptime value only accounted for the `hh:mm:ss` format, but not the `NwNd` format. I added that to the `UPTIME` regex as well.

**Comment by FragmentedPacket** [comment]
@brandomando Do you mind uploading a separate **.raw** and **.parsed** files for the tests rather than editing the existing one? After that change and tests passing, I will get this merged in.

File: templates/cisco_asa_show_route.template [hunk 1]
```diff
@@ -6,7 +6,7 @@ Value DISTANCE (\d+)
 Value METRIC (\d+)
 Value NEXTHOPIP (\d+\.\d+\.\d+\.\d+)
 Value NEXTHOPIF (\S+)
-Value UPTIME (\d+:\d+:\d+)
+Value UPTIME ((\d+:\d+:\d+)|(\d+w\d+d))
```
Comment (jmcgill298) [inline templates/cisco_asa_show_route.template]: Does `\S+` work for UPTIME? I'm sure this converts to `\d+y\d+d` after a year.

**Pull request merged**